File: leetcode/ransom-note.go
package main

func canConstruct(ransomNote string, magazine string) bool {
	ransomNoteCount := make([]int, 26)
	magazineCount := make([]int, 26)

	for i := 0; i < len(ransomNote); i++ {
		aa := ransomNote[i]
		val := int(aa) - int('a')
		ransomNoteCount[val]++
	}
	for i := 0; i < len(magazine); i++ {
		aa := magazine[i]
		val := int(aa) - int('a')
		magazineCount[val]++
	}

	for i := 0; i < 26; i++ {
		if ransomNoteCount[i] > magazineCount[i] {
			return false
		}
	}
	return true
}

//func main() {
//
//}

File: leetcode/ransom-note.py
from typing import List


class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        dictionary = {}
        for ch in magazine:
            if ch in dictionary:
                dictionary[ch] += 1
            else:
                dictionary[ch] = 1
        for ch in ransomNote:
            if ch in dictionary:
                dictionary[ch] -= 1
                if dictionary[ch] == 0:
                    del dictionary[ch]
            else:
                return False
        return True


# inputs = ["a", "b"]
inputs = ["aa", "ab"]
inputs = ["aa", "aab"]
sol = Solution().canConstruct(inputs[0], inputs[1])
print(sol)
